### PR TITLE
Restructure automated-role-assignment sample

### PR DESCRIPTION
**Pull request by ayeshLK**
## Examples
N/A

## Checklist
- [ ] ~Linked to an issue~
- [ ] ~Updated the changelog~
- [ ] ~Added tests~
- [ ] ~Updated the spec~
- [ ] ~Checked native-image compatibility~

### CODE REVIEW

File: examples/automated-role-assignment/main.bal [hunk 1]
```diff
@@ -14,85 +14,109 @@
...
+        content: "React to this message with the following emojis to get your roles:\n\n🔴 - Role1\n🔵 - Role2\n🟢 - Role3"
     };
 
-    http:Request request = new;
-    request.setPayload(message);
-    request.addHeader("Authorization", "Bot BOT_TOKEN");
-    request.addHeader("Content-Type", "application/json");
+    discord:MessageResponse|error createMessageResponse = discord->/channels/[CHANNEL_ID]/messages.post(headers, payload);
+    if createMessageResponse is error {
+        log:printError("Error creating a message: ", createMessageResponse);
```
Comment (NipunaRanasinghe) [inline examples/automated-role-assignment/main.bal]:
```suggestion
        log:printError("Error sending the message: ", createMessageResponse);
```

File: examples/automated-role-assignment/main.bal [hunk 2]
```diff
@@ -14,85 +14,109 @@
...
+        return;
+    }
 
-    http:Response response = check discord->post("/channels/762316177417961504/messages", request);
-    io:println("Role Assignment Message Sent: ", response.getJsonPayload());
+    task:JobId _ = check task:scheduleJobRecurByFrequency(new RoleAssignmentJob(createMessageResponse.id), 60 * 60 * 24);
```
Comment (NipunaRanasinghe) [inline examples/automated-role-assignment/main.bal]: I'm just thinking whether we should keep assigning and changing the roles periodically or, should it be just a one time thing (say we give them one day and assign the roles after that based on the reactions). wdyt?

Comment (ayeshLK) [reply examples/automated-role-assignment/main.bal]: +1

Comment (ayeshLK) [reply examples/automated-role-assignment/main.bal]: Fixed with [97644c0be36988478a3293c5385994f8e8c5e1a4](https://github.com/KamalikaSene/module-ballerinax-discord/pull/3/commits/97644c0be36988478a3293c5385994f8e8c5e1a4)

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Pull request merged**